start_bundle_coupled.R

```diff
@@ -260,61 +267,65 @@ for(scen in common){
   if(is.null(qos) | is.na(qos)) qos <- "short"           # if qos could not be found in scenarios_coupled use short/medium
   start_iter_first <- 1                                  # iteration to start the coupling with
 
-  # look whether there is already a REMIND run (check for old name if provided)
   # Check for existing REMIND and MAgPIE runs and whether iteration can be continued from those (at least one REMIND iteration has to exist!)
-  path_report_found <- NULL
-  start_magpie <- FALSE
+  # Look whether there is already a fulldata.gdx from a former REMIND run (check for old name if provided)
   iter_rem <- 0
-  needle <- scen
-  suche <- paste0(path_remind, "output/", prefix_runname, needle, "-rem-*/REMIND_generic_", prefix_runname, needle, "-rem-*.mif")
-  possibleRemindReport <- mixedsort(Sys.glob(suche))[1]
-  already_rem <- sub(paste0("REMIND_generic_", prefix_runname, needle, "-rem-.*\\.mif"), "fulldata.gdx", possibleRemindReport)
+  suche <- paste0(path_remind, "output/", prefix_runname, scen, "-rem-*/fulldata.gdx")
+  already_rem <- mixedsort(Sys.glob(suche))[1]
+
   if (! is.na(already_rem)) {
-    if (! file.exists(already_rem)) stop(possibleRemindReport, " exists, but ", already_rem, " not!")
     iter_rem <- as.integer(sub(".*rem-(\\d.*)/.*","\\1", already_rem))
-  } else if (! is.na(scenarios_coupled[scen, "oldrun"])) {
+  } else {
     message("Nothing found for ", suche, ", continue with oldrun.")
-    if (isTRUE(stringr::str_sub(scenarios_coupled[scen, "oldrun"], -14, -1) == "/fulldata.gdx") &&
-        file.exists(scenarios_coupled[scen, "oldrun"])) {
-          already_rem <- c(scenarios_coupled[scen, "oldrun"])
+    if (.isFileAndAvailable(scenarios_coupled[scen, "oldrun"], "/fulldata.gdx")) {
+      already_rem <- scenarios_coupled[scen, "oldrun"]
     } else {
-      needle <- scenarios_coupled[scen, "oldrun"]
-      suche <- paste0(path_remind_oldruns, prefix_oldruns, needle, "-rem-*/fulldata.gdx")
+      lookfor <- if (is.na(scenarios_coupled[scen, "oldrun"])) scen else scenarios_coupled[scen, "oldrun"]
+      suche <- paste0(path_remind_oldruns, prefix_oldruns, lookfor, "-rem-*/fulldata.gdx")
       already_rem <- mixedsort(Sys.glob(suche))[1]
     }
   }
-  if (is.na(already_rem)) {
-    message("Nothing found for ", suche, ", starting with ", runname, "-rem-1.")
-  } else {
+  
+  if (.isFileAndAvailable(settings_remind[scen, "path_gdx"], "/fulldata.gdx")) {
```

Comment:
you could put that right at the beginning, right? Only if this is not the case, try to find `already_rem`. Which also avoids cluttering the log with "haven't found anything" messages.

Reply:
Good question and welcome to applied philosophy of coupling. I also considered this, but then had the following case in mind, where you would still need the search for old runs to yield an iteration number that can be continued: a REMIND iteration went infeasible and you want to repeat it and continue the iteration. One solution could be to provide a gdx from a different REMIND run than the direct iteration-predecessor but take the MAgPIE report from the regular preceeding MAgPIE run and continue the iteration counting. Then you wouldn't want to skip the seach for former runs. Is this described in an understandable way?

Reply:
Yes, I got it. The number of use cases is simply annoyingly big :)

Reply:
Oh yeah!